### PR TITLE
Merge branch '2.4' of https://github.com/confluentinc/kafka into 2.4

### PR DESCRIPTION
**Pull request by steverod**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Can you please check previous merges for how we describe the PR, etc?

File: tests/kafkatest/__init__.py [hunk 1]
```diff
@@ -23,3 +23,4 @@
 #
 # For example, when Kafka is at version 1.0.0-SNAPSHOT, this should be something like "1.0.0
 __version__ = '5.4.2.dev0'
+
```
Comment (ijuma) [inline tests/kafkatest/__init__.py]: Is this intentional?

Comment (steverod) [reply tests/kafkatest/__init__.py]: Nope. Removing.

**Comment by steverod** [comment]
> Can you please check previous merges for how we describe the PR, etc?

Changed description to match the standard Jenkins PR.

**Comment by steverod** [comment]
[CPKAFKA-4661] 
Merge conflicts in the AK->CCS merge due to version dfferences. Manually resolve the version number differences.

**Pull request closed without merging**